### ISSUE TITLE
[FP] dead-path false positives for repo-root-relative refs in nested context files

### ISSUE THREAD

**Issue by kims6305-bjk** [alifurkangokce/driftlint#27]
# [FP] dead-path false positives for repo-root-relative refs in nested context files

## Summary

When a context file (e.g. `SKILL.md`) lives in a **subdirectory** of a nested repo, references that are relative to that nested repo's root are flagged as `dead-path` — even though the files exist. `checkDeadPaths` resolves refs against the scan root and the context file's own directory, but never against the enclosing repo root.

In my workspace, 39 of 44 findings were FPs of this kind, all resolved by adding the nested dirs to `.driftlintrc.json` `ignore`.

## Environment

- driftlint v0.18.0 (built from source, commit `f648475`)
- macOS 26.6.2 (Apple Silicon)

## Minimal reproduction

```
outer/                          ← scan root (has .git)
  sub/                          ← nested repo (has .git, but .git presence is irrelevant)
    .agents/skills/myskill/
      SKILL.md                  ← references `eval/comparison/test-sim/`
    eval/comparison/test-sim/   ← the referenced dir EXISTS here
```

`SKILL.md` content:
```markdown
# My Skill

See `eval/comparison/test-sim/` for results.
```

Running `driftlint` at `outer/` produces:

```
sub/.agents/skills/myskill/SKILL.md
  ✗ [dead-path]:3 `eval/comparison/test-sim/` does not exist. ✎ fixable
      ↳ did you mean `sub/eval/comparison/test-sim`?
```

The file exists at `sub/eval/comparison/test-sim/` — the "did you mean" hint even finds it. But `existsAt` only checks:

```ts
// src/checks/deadPaths.ts:126-131
function existsAt(root: string, fileDir: string, rel: string): boolean {
  return (
    fs.existsSync(path.join(root, rel)) ||           // scan root
    fs.existsSync(path.join(root, fileDir, rel))     // context file's dir
  );
}
```

For `file.path = "sub/.agents/skills/myskill/SKILL.md"` and `rel = "eval/comparison/test-sim"`:
- `outer/eval/comparison/test-sim` → doesn't exist
- `outer/sub/.agents/skills/myskill/eval/comparison/test-sim` → doesn't exist
- `outer/sub/eval/comparison/test-sim` → **exists, but never checked**

## Root cause

`existsAt` resolves relative to `root` (scan root) and `fileDir` (the context file's directory). It never walks up to find the enclosing repo root — i.e., the nearest ancestor directory containing `.git`, or simply the top-level directory of the nested project.

This means any context file in a subdirectory of a nested repo that uses repo-root-relative paths (a very common convention in AGENTS.md / SKILL.md files) will produce false positives.

## Real-world impact

My workspace has several vendored/mirrored repos (e.g. `.upstream-mirror/`, `.research/`). Their `SKILL.md` and `CLAUDE.md` files reference paths like `eval/...`, `src/...`, `tests/...` relative to their own repo root. All produce dead-path FPs. The `foreign-context` collapse catches some (11 of 13 refs unresolved → collapsed), but files with fewer refs or a mix of resolved/unresolved still emit individual errors.

## Suggested fix

In `existsAt` (or `checkDeadPaths`), also try resolving `rel` against the **enclosing repo root** — the nearest ancestor of `fileDir` that contains a `.git` directory, or the first path segment boundary. Something like:

```ts
function existsAt(root: string, fileDir: string, rel: string): boolean {
  if (fs.existsSync(path.join(root, rel))) return true;
  if (fs.existsSync(path.join(root, fileDir, rel))) return true;
  // Walk up from fileDir to find enclosing repo root (dir containing .git)
  let dir = fileDir;
  while (dir && dir !== ".") {
    if (fs.existsSync(path.join(root, dir, ".git"))) {
      if (fs.existsSync(path.join(root, dir, rel))) return true;
      break;
    }
    dir = path.dirname(dir);
  }
  return false;
}
```

Alternatively, nested `.git` dirs could be excluded from scanning by default (like `node_modules`), or behind a `--nested-repos` flag.

## Workaround

`.driftlintrc.json` with `ignore` entries for each nested dir works, but requires manual enumeration and hides any real issues inside those dirs.

**Comment by alifurkangokce** [comment]
Reproduced both shapes, fixed, and shipped in [v0.19.0](https://github.com/alifurkangokce/driftlint/releases/tag/v0.19.0). Thank you for the file:line diagnosis and the count — "39 of 44" is the part that told me this wasn't an edge case.

**On your proposed fix.** I built your repro, and then a second one from the real-world section of your report: a vendored mirror under `.research/` with no `.git` of its own. The `.git`-ancestor walk resolves the first and not the second, and your report says the `.git` is incidental, so I needed a marker every one of these files already carries. The one that works is the agent config directory in the path itself — `sub/.agents/skills/myskill/SKILL.md` is owned by `sub`, `.research/mirror/.claude/skills/x/SKILL.md` by `.research/mirror`. No filesystem probing, and it lands on the same answer for the case your version covers. (The same derivation already fixed `$CLAUDE_PROJECT_DIR` in nested `settings.json` back in 0.17, so it's now used consistently in both places.)

**One consequence you predicted.** You wrote that `foreign-context` catches some of these but "files with fewer refs or a mix of resolved/unresolved still emit individual errors." That got worse before it got better: resolving more references pushed several files *below* the collapse ratio, so references belonging to the described repo started arriving as individual errors instead of one collapsed warning. On a real 14-file workspace the error count went 7 → 13 on the first pass.

The rule I settled on: a file that resolves anything against its own nested project root **is** a nested project's document, so what it still can't resolve is that project's surroundings rather than drift in the repo being scanned. Those findings are warnings now, with a hint naming the nested project. Same workspace: errors back to 7, two packages went from a vague "7 of 8 don't resolve" to completely clean, and the rest are specific instead of collapsed.

**Your workaround can go.** `ignore` entries for the nested dirs were hiding anything real inside them; with this you should be able to drop them and still see a quiet report. If any of those 39 come back, or if the warning downgrade turns out to hide something you wanted as an error, please reopen — five regression tests cover this now but your layout is more varied than my fixtures.
